src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCountry.java

```diff
@@ -0,0 +1,35 @@
+package org.diego.tutorial.car.databases.jpa;
+
+import javax.ejb.Stateless;
+import javax.persistence.NoResultException;
+import javax.persistence.TypedQuery;
+
+import org.diego.tutorial.car.model.Country;
+
+/**
+ * Implementation of the JPA persistence with specific
+ * country methods
+ */
+@Stateless
+public class JPAImplCountry extends JPAImpl {
+
+	/**
+	 * Checks if a country exists
+	 * @param countryName Name of the country to check
+	 * @return Boolean
+	 */
+	public boolean countryAlreadyExists(String countryName) {
+		String query = "SELECT country "
+					+ "FROM Country country "
+					+ "WHERE country.countryName='" + countryName + "'";
+		TypedQuery<Country> createQuery = em.createQuery(query, Country.class);
+		
```

Comment:
I would do this better using criteriaQuery

Reply:
done